### PR TITLE
Update object_store version to 0.5.2 and add CHANGELOG

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

re https://github.com/apache/arrow-rs/issues/3229

Rendered changelog: https://github.com/alamb/arrow-rs/blob/alamb/prepare_object_store_0.5.2/object_store/CHANGELOG.md

I double checked the changelog against the list of PRs is https://github.com/apache/arrow-rs/pulls?q=is%3Apr+is%3Aclosed+label%3Aobject-store+  and I think it now correctly reflects everything that was done.

I'll tweak it a bit more immediately prior to the release

### CODE REVIEW

File: object_store/dev/release/update_change_log.sh [hunk 1]
```diff
@@ -49,8 +49,8 @@ docker run -it --rm -e CHANGELOG_GITHUB_TOKEN="$CHANGELOG_GITHUB_TOKEN" -v "$(pw
     --cache-file=.githubchangeloggenerator.cache \
     --cache-log=.githubchangeloggenerator.cache.log \
     --http-cache \
-    --max-issues=300 \
-    --exclude-tags-regex "^\d+\.\d+\.\d+$" \
+    --max-issues=600 \
```
Comment (alamb) [inline object_store/dev/release/update_change_log.sh:52]: need to fetch 600 issues to get some of the older ones that have been resolved!

File: object_store/dev/release/update_change_log.sh [hunk 2]
```diff
@@ -49,8 +49,8 @@ docker run -it --rm -e CHANGELOG_GITHUB_TOKEN="$CHANGELOG_GITHUB_TOKEN" -v "$(pw
...
+    --exclude-tags-regex "(^\d+\.\d+\.\d+$)|(rc)" \
```
Comment (alamb) [inline object_store/dev/release/update_change_log.sh:53]: Also I needed to exclude the 'rc' tags so it didn't make changelog entries for `26.0.0_rc1`

**Review by tustvold — APPROVED** [review]
It would be awesome for https://github.com/apache/arrow-rs/pull/3256 to make this release, but otherwise this looks good

**Comment by alamb** [comment]
> It would be awesome for https://github.com/apache/arrow-rs/pull/3256 to make this release, but otherwise this looks good

It is in!

**Pull request merged**